peerconnection.go

```diff
@@ -2097,6 +2114,19 @@ func (pc *PeerConnection) Close() error {
 	// https://www.w3.org/TR/webrtc/#dom-rtcpeerconnection-close (step #11)
 	pc.updateConnectionState(pc.ICEConnectionState(), pc.dtlsTransport.State())
 
+	// non-canon steps
+	pc.sctpTransport.lock.Lock()
```

Comment:
@Sean-Der I moved these steps lower. I also made Close not block since it's not clear to me if that's allowed per the spec.